[cli] Don't error when user cannot acces pac ns

chmouel opened the pull request:
when we have issue accessing the pac-configmap or listing the repo CR we
would error out. While it's perfectly valid to not be able to have those
right as user.

cf #780

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

savitaashture committed:
Add condition check to verify secret-github-app-token-scoped enabled or not

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

savitaashture commented on pkg/cmd/tknpac/create/repository.go:
```diff
@@ -76,15 +76,18 @@ func repositoryCommand(run *params.Run, ioStreams *cli.IOStreams) *cobra.Command
 			if err != nil {
-				return err
+				// if we don't have access to the pipelines as code info configmap (ie acl, don't let you)
+				// we won't try to update it
```
How about adding some info log here  :thinking: 
Because if we just generate template and return how end user will get to know what happened

chmouel replied:
i don't think that log would be useful, this would show on the command line everytime and as a user who is not admin they would not be able to do anything about it....

savitaashture commented on pkg/cmd/tknpac/create/repository.go:
```diff
@@ -76,15 +76,18 @@ func repositoryCommand(run *params.Run, ioStreams *cli.IOStreams) *cobra.Command
 			}
 
 			if pacInfo.IsGithubAppInstalled(ctx, run, installationNS) {
-				if strings.Contains(createOpts.Event.URL, "github") {
```
I remember that this if condition added to cover scenario like 

GitHubApp is installed but event.URL is for GitLab or BitBucket then it won't proceed for Webhook creation part that's why added this condition to make sure GitHubApp is installed and event URL is GitHub then just return and don't go for Webhook creation

chmouel replied:
this is pretty buggy check, what about the gitea/bitbucket and others that somehow has a github string into the url?

moreover we don't have any unit tests for this so we may as well guess what's right or not...

savitaashture replied:
> what about the gitea/bitbucket and others that somehow has a github string into the url

agree

then how do we ensure this part otherwise UX will change for create repository

chmouel replied:
that's why there was a switch before 🙃

chmouel replied:
and this calls as well will fail on GHE i think too

savitaashture replied:
But we have to handle this somehow or else functionality will break

I mean If GitHubApp is installed then it won't allow for webhook creation at all for other provider

savitaashture replied:
How about adding check here https://github.com/openshift-pipelines/pipelines-as-code/blob/26a61ccc270d864523c01c21fcdd9127b212bfc3/pkg/cmd/tknpac/create/repository.go#L93 based on the providerName

I mean If GitHubApp is installed but provider name is other then Github proceed for webhook creation if not just don't got for webhook creation

WDYT?

chmouel replied:
so what is worse? not allowing user to generate pipelinerun as non cluster admin for github apps or don't have the flow continue for other providers that is much less used ?

you can still create webhook creation with the tkn pac switch directly....

chmouel replied:
what about you fix it in a following pr :D ?

savitaashture replied:
Yes we can still create webhook using
`tkn-pac webhook add` for all 3 provider (GitHub, GitLab, Bitbucket) even though GitHubApp is configured

Only thing is UX will change with this PR

chmouel replied:
i think at some point we can just do tkn pac bootstrap github-webhook/gitlab/bitbucket/github-app and not care about being too smart with one command for everything

savitaashture replied:
Yes we can do that UX refactoring and change commands but that will take its own time as PAC is GA now and if we change some command we need to provide backward compatibility also

savitaashture replied:
> this is pretty buggy check, what about the gitea/bitbucket and others that somehow has a github string into the url?

> moreover we don't have any unit tests for this so we may as well guess what's right or not...

generally other SCM should not have github in there URL even if user provides that intentionally then its there responsibility i would say

savitaashture replied:
How about keeping this check as it is and adding `To Do` comment  in the code for now (Though it won't handle fully but basic check it does)

chmouel committed:
[cli] Don't error when user cannot acces pac ns

when we have issue accessing the pac-configmap or listing the repo CR we
would error out. While it's perfectly valid to not be able to have those
right as user.

cf #780

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>
